frontend/public/components/ThemeProvider.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { useUserSettings } from '@console/shared';
+
+export const THEME_USER_SETTING_KEY = 'console.theme';
+export const THEME_LOCAL_STORAGE_KEY = 'bridge/theme';
+const THEME_SYSTEM_DEFAULT = 'systemDefault';
+const THEME_DARK_CLASS = 'pf-theme-dark';
+const THEME_DARK = 'dark';
+
+export const updateThemeClass = (htmlTagElement: HTMLElement, theme: string) => {
+  let systemTheme: string;
+  if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+    systemTheme = THEME_DARK;
+  }
+  if (theme === THEME_DARK || (theme === THEME_SYSTEM_DEFAULT && systemTheme === THEME_DARK)) {
+    htmlTagElement.classList.add(THEME_DARK_CLASS);
+  } else {
+    htmlTagElement.classList.remove(THEME_DARK_CLASS);
+  }
+};
+
+export const ThemeContext = React.createContext<string>('');
+
+export const ThemeProvider = ({ children }) => {
+  const htmlTagElement = document.documentElement;
+  const localTheme = localStorage.getItem(THEME_LOCAL_STORAGE_KEY);
+  const [theme, , themeLoaded] = useUserSettings(
+    THEME_USER_SETTING_KEY,
+    THEME_SYSTEM_DEFAULT,
+    true,
+  );
+  const mqListener = React.useCallback(
+    (e) => {
+      if (e.matches) {
+        htmlTagElement?.classList.add(THEME_DARK_CLASS);
+      } else {
+        htmlTagElement?.classList.add(THEME_DARK_CLASS);
+      }
+    },
+    [htmlTagElement],
+  );
+  React.useEffect(() => {
+    const darkThemeMq = window.matchMedia('(prefers-color-scheme: dark)');
+    if (theme === THEME_SYSTEM_DEFAULT) {
+      darkThemeMq.addListener(mqListener);
```

Comment:
`addListener` is deprecated, see also https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener

It was also highlighted in my VScode, can we use the recommended alternative here?

![image](https://user-images.githubusercontent.com/139310/159897394-dd6a9ec1-1c28-4750-9363-8fb9e2d04865.png)


```suggestion
      darkThemeMq.addEventListener('change', mqListener);
```